### PR TITLE
[FEAT] Add GeoLocation constructor from EdgePoint

### PR DESCRIPTION
**Pull request by jarodlam**
Add a method to easily get the coordinates of an `EdgePoint`.

```julia
julia> ep = EdgePoint(6913853675, 144699879, 0.5)  # Half-way between nodes 6913853675 and 144699879
julia> GeoLocation(g, ep)
GeoLocation(-37.81912355, 144.9580393, 0.0)
```

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+# Fields
 - `lat::Float64`: Latitude.
 - `lon::Float64`: Longitude.
 - `alt::Float64`: Altitude.
+
+# Constructors
+    GeoLocation(lat, lon)
+    GeoLocation(point::Vector{<:Real})
+    GeoLocation(point_vector::Vector{<:Vector{<:Real}})
+    GeoLocation(g::OSMGraph, ep::EdgePoint)
```
Comment (jarodlam) [inline src/types.jl:13]: Add documentation for GeoLocation constructors

File: src/types.jl [hunk 2]
```diff
@@ -20,6 +26,9 @@ GeoLocation(point_vector::Vector{<:Vector{<:Real}})::Vector{GeoLocation} = [GeoL
 function Base.:(==)(loc1::GeoLocation, loc2::GeoLocation)
     return loc1.lat == loc2.lat && loc1.lon == loc2.lon && loc1.alt == loc2.alt
 end
+function Base.isapprox(loc1::GeoLocation, loc2::GeoLocation)
+    return loc1.lat ≈ loc2.lat && loc1.lon ≈ loc2.lon && loc1.alt ≈ loc2.alt
+end
```
Comment (jarodlam) [inline src/types.jl:31]: Add method for approximate equality of GeoLocations

File: src/types.jl [hunk 3]
```diff
@@ -196,3 +211,11 @@ abstract type DijkstraDict <: Dijkstra end
...
 abstract type AStarVector <: AStar end
 abstract type AStarDict <: AStar end
+
+"""
+Additional GeoLocation methods that require types defined above.
+"""
+GeoLocation(g::OSMGraph, ep::EdgePoint)::GeoLocation = GeoLocation(
+    lon = g.nodes[ep.n1].location.lon + (g.nodes[ep.n2].location.lon - g.nodes[ep.n1].location.lon) * ep.pos,
+    lat = g.nodes[ep.n1].location.lat + (g.nodes[ep.n2].location.lat - g.nodes[ep.n1].location.lat) * ep.pos
+)
```
Comment (jarodlam) [inline src/types.jl:221]: This is at the bottom of the file because it requires the declaration of `OSMGraph`.

File: src/graph.jl [hunk 4]
```diff
@@ -55,7 +55,7 @@ function graph_from_object(osm_data_object::Union{XMLDocument,Dict};
         g.dijkstra_states = Vector{Vector{U}}(undef, length(g.nodes))
     end
 
-    add_kdtree!(g)
+    add_kdtree_and_rtree!(g)
```
Comment (jarodlam) [inline src/graph.jl:58]: Bug fix. The R-tree wasn't getting added to the `OSMGraph` object, so `nearest_way` wasn't working.

**Review by bjmommers — APPROVED** [review]
Looks good, approved.

**Pull request merged**